### PR TITLE
[pod-gateway] Fix issue where values.yaml defined 'configFile' wouldn't work in vpn addon

### PR DESCRIPTION
**Pull request by jmmaloney4**
**Description of the change**

Because the default values.yaml file has `configFileSecret` set to `openvpn` by default, it is impossible to use the built in `configFile` with no secret. If you attempt to override `configFileSecret` to `null` the default value takes precedence, and the gateway will not start because of the lack of `openvpn` secret. 

**Benefits**

**Possible drawbacks**

**Applicable issues**

- fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [x] Variables are documented in the README.md (this can be done with using our helm-docs wrapper `./hack/gen-helm-docs.sh stable <chart>`)

### CODE REVIEW

**Comment by bjw-s** [comment]
In my local testing, when you specify the following values you should be able to run without the configFile:

```yaml
addons:
  vpn:
    enabled: true
    type: openvpn
    openvpn:
    configFileSecret: 
```

Although I do think it would be a bit cleaner if some of these keys are not set here but are left to the user's preference. Unfortunately I ran into a small bug in the common library when trying this, so that would need to get fixed before we can remove the `openvpn` and `wireguard` keys.

**Comment by angelnu** [comment]
Thanks for the PR!

The values I picked as the defaults were heavily influenced by me using openvpn - so we can remove any non-default values that cannot be overwritten by the user.

Is there anything to remove beyond `configFileSecret` and potentially `type` ? Is there any issue with the null values @bjw-s or are they merged with the common defaults?

**Comment by bjw-s** [comment]
> Is there anything to remove beyond `configFileSecret` and potentially `type` ? Is there any issue with the null values @bjw-s or are they merged with the common defaults?

By specifying them as null-values, the default values seem to get overwritten with `null` (I still can't really figure out why :X), potentially causing the chart to render incorrectly. That's why I proposed to remove the `wireguard` and `openvpn` keys.

File: charts/stable/pod-gateway/values.yaml [hunk 1]
```diff
@@ -76,7 +76,7 @@ addons:
     openvpn:
```
Comment (angelnu) [inline charts/stable/pod-gateway/values.yaml:76]:
```suggestion
    # openvpn:
```

File: charts/stable/pod-gateway/values.yaml [hunk 2]
```diff
@@ -76,7 +76,7 @@ addons:
...
     wireguard:
```
Comment (angelnu) [inline charts/stable/pod-gateway/values.yaml:77]:
```suggestion
    # wireguard:
```

File: charts/stable/pod-gateway/values.yaml [hunk 3]
```diff
@@ -76,7 +76,7 @@ addons:
...
     env:
```
Comment (angelnu) [inline charts/stable/pod-gateway/values.yaml:78]:
```suggestion
    # env:
```

**Review by angelnu — CHANGES_REQUESTED** [review]

**Comment by angelnu** [comment]
@jmmaloney4 - please merge the suggestions when you have the chance so we can merge your PR. Thanks again for submitting 👍

**Comment by angelnu** [comment]
@jmmaloney4 - would you be able to add the docs? If you need any help please let me know - I can, if necessary, add the doc to your PR so we can merge this fix.

**Comment by jmmaloney4** [comment]
@angelnu is this change sufficient?

**Comment by angelnu** [comment]
It looks good @jmmaloney4 - only step left is to re-generate the README.md.

Can you please run `task chart:helm-docs CHART_TYPE=stable CHART=pod-gateway` in your PR? Then we are ready to merge :-)

**Comment by jmmaloney4** [comment]
@angelnu should be good to go.

**Review by angelnu — APPROVED** [review]

**Comment by angelnu** [comment]
Thanks @jmmaloney4 - now we need an owner to merge.

**Review by bjw-s — APPROVED** [review]
Approved! Thanks a lot for the contribution!!

**Comment by bjw-s** [comment]
@all-contributors please add @jmmaloney4 for code

**Pull request merged**